### ISSUE TITLE
Failed to convert cloud formation to terraform - KeyError: 'Properties'

### ISSUE THREAD

**Issue by sanjuthomas** [DontShaveTheYak/cf2tf#50]
Input file - https://github.com/awslabs/amazon-ecs-nodejs-microservices/blob/master/3-microservices/infrastructure/ecs.yml

I cloned the repo to the local box and tried the following - 

**cf2tf ecs.yml > main.tf**
// Converting ecs.yml to Terraform!
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.10/bin/cf2tf", line 8, in <module>
    sys.exit(cli())
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/cf2tf/app.py", line 44, in cli
    config = TemplateConverter(cf_template, search_manger).convert()
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/cf2tf/convert.py", line 71, in convert
    tf_resources = self.convert_to_tf(self.manifest)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/cf2tf/convert.py", line 123, in convert_to_tf
    tf_resources.extend(converter(resources))
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/cf2tf/convert.py", line 302, in convert_resources
    resource_props["Properties"], functions.ALL_FUNCTIONS
KeyError: 'Properties'

**cat main.tf**
// Cloning Terraform src code to /var/folders/y3/bssrxrxd3811hxq8zzct1j700000gn/T/terraform_src...
 code has been checked out.

**Comment by shadycuz** [comment]
Hey thanks for opening this issue.

I see the problem here

```
InternetGateway:
    Type: AWS::EC2::InternetGateway
```

I thought that all resources had to have a properties section but I guess not.

I'll try to fix this tomorrow.

**Comment by roboweaver** [comment]
Not sure if this is the same issue, but my error has the same error about AWS::StackId

` // Converting vpc.yaml to Terraform!
// Cloning Terraform src code to /var/folders/1f/fmjsp9115c12wxfnkbh_2bm40000gn/  [####--------------------------------]   12%  00:00:40 code has been checked out.
Traceback (most recent call last):
  File "/usr/local/bin/cf2tf", line 8, in <module>
    sys.exit(cli())
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/cf2tf/app.py", line 44, in cli
    config = TemplateConverter(cf_template, search_manger).convert()
  File "/usr/local/lib/python3.10/site-packages/cf2tf/convert.py", line 71, in convert
    tf_resources = self.convert_to_tf(self.manifest)
  File "/usr/local/lib/python3.10/site-packages/cf2tf/convert.py", line 123, in convert_to_tf
    tf_resources.extend(converter(resources))
  File "/usr/local/lib/python3.10/site-packages/cf2tf/convert.py", line 301, in convert_resources
    resolved_values = self.resolve_values(
  File "/usr/local/lib/python3.10/site-packages/cf2tf/convert.py", line 155, in resolve_values
    data[key] = self.resolve_values(value, allowed_func, prev_func)
  File "/usr/local/lib/python3.10/site-packages/cf2tf/convert.py", line 169, in resolve_values
    return [self.resolve_values(item, allowed_func, prev_func) for item in data]
  File "/usr/local/lib/python3.10/site-packages/cf2tf/convert.py", line 169, in <listcomp>
    return [self.resolve_values(item, allowed_func, prev_func) for item in data]
  File "/usr/local/lib/python3.10/site-packages/cf2tf/convert.py", line 155, in resolve_values
    data[key] = self.resolve_values(value, allowed_func, prev_func)
  File "/usr/local/lib/python3.10/site-packages/cf2tf/convert.py", line 152, in resolve_values
    return functions.ref(self, value)
  File "/usr/local/lib/python3.10/site-packages/cf2tf/conversion/expressions.py", line 767, in ref
    return handle_pseduo_var(template, var_name)
  File "/usr/local/lib/python3.10/site-packages/cf2tf/conversion/expressions.py", line 815, in handle_pseduo_var
    raise ValueError(f"Unable to process pseudo var {pseudo_name}.")
ValueError: Unable to process pseudo var AWS::StackId.
`

**Comment by roboweaver** [comment]
Only think I can see in the stack that even uses that variable are tags as in this:

```
  PublicNetworkAcl:
    Type: AWS::EC2::NetworkAcl
    Properties:
      VpcId: !Ref VPC
      Tags:
        - Key: Application
          Value: !Ref AWS::StackId
        - Key: Network
          Value: Public
        - Key: Name
          Value: PublicNACL
```

**Comment by roboweaver** [comment]
FWIW - commenting out those tags resulted in a successful run, so it definitely wasn't grokking the AWS::StackId variable.

**Comment by shadycuz** [comment]
@roboweaver The stackID is not supported because there is no direct equivalent in Terraform. If you create a new issue perhaps we can discuss alternatives ways to implement stackID in Terraform.

**Comment by roboweaver** [comment]
No worries, I realized after posting that it was probably silly to try and use that for exactly the reason you point out.

**Comment by shadycuz** [comment]
@roboweaver well it is valid Cloudformation and it's a problem we will have to tackle eventually.

**Comment by shadycuz** [comment]
@roboweaver There was already a discussion around StackID in #22, I will try and implement a fix shortly.

**Comment by shadycuz** [comment]
@sanjuthomas The KeyError has been fixed in [v0.2.7](https://github.com/DontShaveTheYak/cf2tf/releases/tag/0.2.7). 🎉 

Please open an issue if you see any other errors, thanks!

**Comment by sanjuthomas** [comment]
Thanks, @shadycuz
